// @scalar/docusaurus/dist/index.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const vitest_1 = require("vitest");
const index_1 = __importDefault(require("./index"));
(0, vitest_1.describe)('ScalarDocusaurus', () => {
    (0, vitest_1.describe)('plugin creation', () => {
        (0, vitest_1.it)('returns a plugin object with required methods', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
            });
            (0, vitest_1.expect)(plugin).toHaveProperty('name', '@scalar/docusaurus');
            (0, vitest_1.expect)(plugin).toHaveProperty('injectHtmlTags');
            (0, vitest_1.expect)(plugin).toHaveProperty('loadContent');
            (0, vitest_1.expect)(plugin).toHaveProperty('contentLoaded');
        });
    });
    (0, vitest_1.describe)('injectHtmlTags', () => {
        (0, vitest_1.it)('injects default CDN script when no CDN specified', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
            });
            const result = plugin.injectHtmlTags();
            (0, vitest_1.expect)(result).toEqual({
                preBodyTags: [
                    {
                        tagName: 'script',
                        attributes: {
                            src: 'https://cdn.jsdelivr.net/npm/@scalar/api-reference',
                        },
                    },
                ],
            });
        });
        (0, vitest_1.it)('injects custom CDN script when specified', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
                cdn: 'https://cdn.example.com/custom-scalar.js',
            });
            const result = plugin.injectHtmlTags();
            (0, vitest_1.expect)(result).toEqual({
                preBodyTags: [
                    {
                        tagName: 'script',
                        attributes: {
                            src: 'https://cdn.example.com/custom-scalar.js',
                        },
                    },
                ],
            });
        });
    });
    (0, vitest_1.describe)('loadContent', () => {
        (0, vitest_1.it)('returns default options with merged configuration', async () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Custom Label',
                route: '/custom-route',
                configuration: {
                    theme: 'purple',
                },
            });
            const content = await plugin.loadContent();
            (0, vitest_1.expect)(content).toEqual({
                label: 'Custom Label',
                route: '/custom-route',
                showNavLink: true,
                configuration: {
                    _integration: 'docusaurus',
                    theme: 'purple',
                },
            });
        });
        (0, vitest_1.it)('applies default values for missing options', async () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Test Label',
                route: '/test-route',
            });
            const content = await plugin.loadContent();
            (0, vitest_1.expect)(content).toEqual({
                label: 'Test Label',
                route: '/test-route',
                showNavLink: true,
                configuration: {
                    _integration: 'docusaurus',
                },
            });
        });
        (0, vitest_1.it)('merges configuration objects correctly', async () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Test',
                route: '/test',
                configuration: {
                    theme: 'kepler',
                    layout: 'modern',
                },
            });
            const content = await plugin.loadContent();
            (0, vitest_1.expect)(content?.configuration).toEqual({
                _integration: 'docusaurus',
                theme: 'kepler',
                layout: 'modern',
            });
        });
    });
    (0, vitest_1.describe)('contentLoaded', () => {
        (0, vitest_1.it)('adds navbar link with baseUrl when showNavLink is true', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/my-site/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'API Docs',
                route: '/api',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items).toHaveLength(1);
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0]).toEqual({
                to: '/my-site/api',
                label: 'API Docs',
                position: 'left',
            });
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/my-site/api',
            }));
        });
        (0, vitest_1.it)('adds navbar link with default values', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/docs/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0]).toEqual({
                to: '/docs/scalar',
                label: 'Scalar',
                position: 'left',
            });
        });
        (0, vitest_1.it)('does not add navbar link when showNavLink is false', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
                showNavLink: false,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items).toHaveLength(0);
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalled();
        });
        (0, vitest_1.it)('handles root baseUrl correctly', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                showNavLink: true,
                route: '/scalar',
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0].to).toBe('/scalar');
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/scalar',
            }));
        });
        (0, vitest_1.it)('always adds route regardless of showNavLink setting', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/base/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/api',
                showNavLink: false, // navbar disabled
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            // Route should still be added
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/base/api',
                exact: true,
            }));
            // But navbar should not have items
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items).toHaveLength(0);
        });
        (0, vitest_1.it)('passes content to route configuration', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
            });
            const testContent = {
                configuration: {
                    theme: 'purple',
                },
            };
            plugin.contentLoaded?.({
                content: testContent,
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/scalar',
                exact: true,
                configuration: {
                    theme: 'purple',
                },
            }));
        });
        (0, vitest_1.it)('uses default route when none specified', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/site/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'Scalar',
                route: '/scalar',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0].to).toBe('/site/scalar');
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/site/scalar',
            }));
        });
        (0, vitest_1.it)('uses default label when none specified', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                route: '/api',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0].label).toBe('Scalar');
        });
    });
    (0, vitest_1.describe)('edge cases', () => {
        (0, vitest_1.it)('handles complex baseUrl paths', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/very/deep/nested/path/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'API',
                route: '/docs',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0].to).toBe('/very/deep/nested/path/docs');
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/very/deep/nested/path/docs',
            }));
        });
        (0, vitest_1.it)('handles empty route string', () => {
            const mockContext = {
                siteConfig: {
                    baseUrl: '/base/',
                    themeConfig: {
                        navbar: {
                            items: [],
                        },
                    },
                },
            };
            const mockActions = {
                addRoute: vitest_1.vi.fn(),
            };
            const plugin = (0, index_1.default)(mockContext, {
                label: 'API',
                route: '',
                showNavLink: true,
            });
            plugin.contentLoaded?.({
                content: {},
                actions: mockActions,
            });
            (0, vitest_1.expect)(mockContext.siteConfig.themeConfig.navbar.items[0].to).toBe('/base/');
            (0, vitest_1.expect)(mockActions.addRoute).toHaveBeenCalledWith(vitest_1.expect.objectContaining({
                path: '/base/',
            }));
        });
    });
});
