// @scalar/docusaurus/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const node_path_1 = __importDefault(require("node:path"));
const utils_1 = require("@docusaurus/utils");
/**
 * Used to set default options from the user-provided options
 * This is also useful to ensure backwards compatibility with older configs that don't have the new options
 */
const createDefaultScalarOptions = (options) => ({
    showNavLink: true,
    ...options,
    configuration: {
        _integration: 'docusaurus',
        ...(options.configuration ?? {}),
    },
});
/**
 * Scalar's Docusaurus plugin for Api References
 */
const ScalarDocusaurus = (context, options) => {
    const defaultOptions = createDefaultScalarOptions(options);
    const { baseUrl } = context.siteConfig;
    return {
        name: '@scalar/docusaurus',
        /**
         * Load the Standalone API Reference script
         * This is loaded into the dom once for every plugin thats loaded BUT it only downloads the script once
         */
        injectHtmlTags() {
            return {
                preBodyTags: [
                    {
                        tagName: 'script',
                        attributes: {
                            src: options.cdn ?? 'https://cdn.jsdelivr.net/npm/@scalar/api-reference',
                        },
                    },
                ],
            };
        },
        loadContent() {
            return defaultOptions;
        },
        contentLoaded({ content, actions }) {
            const { addRoute } = actions;
            // If showNavLink is true, add a link to the navbar
            if (defaultOptions.showNavLink) {
                ;
                context.siteConfig.themeConfig.navbar.items.push({
                    to: (0, utils_1.normalizeUrl)([baseUrl, defaultOptions.route ?? '/scalar']),
                    label: defaultOptions.label ?? 'Scalar',
                    position: 'left',
                });
            }
            // Add the appropriate route based on the module system
            addRoute({
                path: (0, utils_1.normalizeUrl)([baseUrl, defaultOptions.route ?? '/scalar']),
                component: node_path_1.default.resolve(__dirname, './ScalarDocusaurus'),
                exact: true,
                ...content,
            });
        },
    };
};
exports.default = ScalarDocusaurus;
